sql/src/test/java/org/apache/druid/sql/calcite/CalciteNestedDataQueryTest.java

```diff
@@ -174,7 +198,72 @@
                     .rows(ROWS)
                     .buildMMappedIndex();
 
-    return new SpecificSegmentsQuerySegmentWalker(conglomerate).add(
+    final QueryableIndex indexMix11 =
+        IndexBuilder.create()
+                    .tmpDir(temporaryFolder.newFolder())
+                    .segmentWriteOutMediumFactory(OffHeapMemorySegmentWriteOutMediumFactory.instance())
+                    .schema(
+                        new IncrementalIndexSchema.Builder()
+                            .withMetrics(
+                                new CountAggregatorFactory("cnt")
+                            )
+                            .withDimensionsSpec(PARSER)
```

Comment:
Maybe make this call with `PARSER.getParseSpec().getDimensionSpec()` instead and avoid usage of the deprecated method?

Reply:
these tests are in the model of the other calcite query tests which still use parser because they predate the newer split dimensionschema and inputsource/inputformat stuff of native batch

I will see if i can rework them to not use outdated stuff, and just make `InputRow` with `MapInputRowParser.parse` like all of the newer `InputFormat` implementations do